src/api.ts

```diff
@@ -1,26 +1,32 @@
 type RequestOptions = {
-    method?: 'GET' | 'POST',
-    queryParams?: Record<'s', string>,
-}
+  method?: "GET" | "POST";
+  queryParams?: { [k: string]: string };
+};
 
 const API_BASE_URL = "http://www.omdbapi.com/";
 
-const requestData = async ({ method, queryParams }: RequestOptions = { method: 'GET' }) => {
-    const params = {
-        apikey: import.meta.env.VITE_API_KEY,
-        ...queryParams,
-    }
+const requestData = async (
```

Comment:
You might want to add path for request, e.g. `requestData('/some/path', { queryParams: { i: id || "" } });`